src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 3;
+    path_parameters.num_segments               = 500;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 9;
+    path_parameters.initial_speed              = 2;
+    path_parameters.final_speed                = 5;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+
+    app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                 &trajectory);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path, path_parameters.t_start, path_parameters.t_end, arc_length_param);
+
+    std::vector<float> velocity =
+        getSpeedsFromTrajectory(&trajectory, path_parameters.initial_speed);
+    std::vector<float> acceleration = getAccelerationsFromSpeed(velocity, &trajectory);
+
+    for (float vel : velocity)
+    {
+        EXPECT_TRUE(vel <= path_parameters.max_allowable_speed * 1.1 &&
+                    vel >= path_parameters.initial_speed);
+    }
+
+    EXPECT_NEAR(velocity.back(), path_parameters.final_speed, 0.1);
+    EXPECT_FLOAT_EQ(velocity.front(), path_parameters.initial_speed);
+
+    for (float acc : acceleration)
+    {
+        EXPECT_TRUE(fabs(acc) <= path_parameters.max_allowable_acceleration * 1.2);
+    }
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.y,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).y,
+        0.01);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.x,
+                shared_polynomial2d_getValueOrder3(path, path_parameters.t_start).x,
+                0.01);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.y,
+                shared_polynomial2d_getValueOrder3(path, path_parameters.t_start).y,
+                0.01);
+}
+
+TEST_F(TrajectoryPlannerTest,
+       check_trajectory_length_and_end_points_match_path_reverse_parameterization)
```

Comment:
Similar to above, can we split this into 2 tests?

Reply:
:+1: done